### PR TITLE
Added UniqueID storing and lookup

### PR DESCRIPTION
**Pull request by jameshartig**
This eliminates the 2 todo's in the README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,7 +10,35 @@ dropped if that email is opted out of that particular email.
 
 ## Sending
 
-SendGrid is used to send emails and they're sent at most once. A `--sendgrid-key` is required.
+SendGrid is used to send emails and they're sent at most once. A
```
Comment (mediocregopher) [inline README.md]: > SendGrid is used to send emails. Emails are sent at most once.
> :sheep:

File: README.md [hunk 2]
```diff
@@ -10,7 +10,35 @@ dropped if that email is opted out of that particular email.
...
+`--sendgrid-key` is required.
+
+## Prerequisites
+
+You must have a SendGrid account. In order to store statistics you must be
+running a MongoDB instance and send the address to `--mongo-addr`. You must also
+publicly expose the postmaster webhook port to the Internet. Do NOT expose the
+RPC port.
+
+In order to provide resisency against the service crashing before it had a
```
Comment (mediocregopher) [inline README.md]: resiliency?

File: db/stats.go [hunk 3]
```diff
@@ -18,34 +20,72 @@ const (
...
 type StatsJob struct {
-	Email     string `json:"email" validate:"email,nonzero"` //Email address of the intended recipient
-	Timestamp int64  `json:"timestamp,omitempty"`
-	Type      string `json:"event" validate:"nonzero"`    //One of: bounce, deferred, delivered, dropped, processed
-	StatsID   string `json:"stats_id" validate:"nonzero"` //json key must match db.UniqueArgID
-	Reason    string `json:"reason,omitempty" validate:"max=1024"`
+	//Email address of the intended recipient
+	Email string `json:"email" validate:"email,nonzero"`
+
+	Timestamp int64 `json:"timestamp,omitempty"`
```
Comment (mediocregopher) [inline db/stats.go]: would it be a giant pain to use `timeutil.Timestamp` here?

File: db/stats.go [hunk 4]
```diff
@@ -18,34 +20,72 @@ const (
...
+	EmailFlags int64 `json:"emailFlags" bson:"ef"`
+
+	// StateFlags represent the current state of the email
+	StateFlags int64 `json:"stateFlags" bson:"s"`
+
+	// UniqueID was the original uniqueID sent to us in rpc.Enqueue
+	UniqueID string `json:"uniqueID" bson:"uid"`
+
+	// TSCreated is the time that the email was sent
+	TSCreated time.Time `json:"tsCreated" bson:"tc"`
```
Comment (mediocregopher) [inline db/stats.go]: possible `timeutil.Timestamp` for these too?

File: rpc/enqueue.go [hunk 5]
```diff
@@ -14,16 +14,11 @@ import (
 	"github.com/levenlabs/postmaster/sender"
 )
 
-// EnqueueArgs defines the arguments for the Postmaster.Enqueue endpoint
-type EnqueueArgs sender.Mail
+type enqueueArgs sender.Mail
```
Comment (mediocregopher) [inline rpc/enqueue.go]: honestly we don't even really have to make an `enqueueArgs`, just use a `*sender.Mail` as args if you want

File: db/stats.go (hunk 3 above)
Comment (fastest963) [inline db/stats.go]: nope, done

**Pull request merged**